### PR TITLE
Add additional checks and negative tests #7

### PR DESCRIPTION
**Pull request by DaryaAndreyuk**
For positive tests before moving to a next registration tab we compare values from Excel and real values that inserted to all input fields.

For negative tests we try to register without mandatory fields or with invalid phone number, fail the registration process and etc. The goal of the test is to assert that error message appeared on the screen.

### CODE REVIEW

File: src/test/java/basecomponents/BaseTest.java [hunk 1]
```diff
@@ -31,5 +40,20 @@ public void tearDown() {
             driver.quit();
         }
     }
+
+    public RegisterGamePage initializeRegisterPage() {
+        landingPage.closeCookieAlert();
+        UpcomingGamesPage upcomingGamesPage = landingPage.loginApplication("marya.masha@yahoo.com", "Ilovemusicquiz!9728");
```
Comment (topsycreed) [inline src/test/java/basecomponents/BaseTest.java:46]: Looks like real login and password, should be moved to property files

File: src/test/java/RegistrationTeamTest.java [hunk 2]
```diff
@@ -5,34 +5,72 @@
...
+        registerGamePage.waitForWebElementToAppear(registerGamePage.getFinalRegButton());
         assertTrue(registerGamePage.getFinalRegButton().isDisplayed() && registerGamePage.getFinalRegButton().isEnabled());
         // registerGamePage.registerTeam();
     }
+
+    @Test
+    public void loginDataInExcelValidation() {
+        LandingPage landingPage = new LandingPage(driver);
+        landingPage.closeCookieAlert();
+        landingPage.loginApplication("marya.masha@yahoo.com", "Ilovemusicquiz!9728");
```
Comment (topsycreed) [inline src/test/java/RegistrationTeamTest.java:45]: Move to property file. Commit should be created as ammend commit with forced push to remove history that you placed real credentials

File: src/test/java/RegistrationTeamTest.java [hunk 3]
```diff
@@ -5,34 +5,72 @@
...
 
 import org.example.utils.ExcelUtils;
 
 public class RegistrationTeamTest extends BaseTest {
 
     @Test
-    public void checkTeamRegistration() {
-
-        Map<String, String> dataMap = ExcelUtils.getExcelDataToMap(ExcelUtils.getPathToResourceFile());
+    public void registerTeamValidation() {
```
Comment (topsycreed) [inline src/test/java/RegistrationTeamTest.java:18]: Add tags for positive, negative, smoke tests and etc. Since we could run not all tests each time

Comment (topsycreed) [reply src/test/java/RegistrationTeamTest.java:18]: You also need to create tasks inside pom.xml for that using maven-surefire-plugin for configuration several test runs. See example for running only tests with @Tag("smoke") annotation without @Tag("defect") tests:
`<build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.2.5</version>
                <configuration>
                    <groups>smoke</groups>
                    <excludedGroups>defect</excludedGroups>
                </configuration>
            </plugin>
        </plugins>
    </build>`

Comment (topsycreed) [reply src/test/java/RegistrationTeamTest.java:18]: See: https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html#filtering-by-tags

File: src/test/java/RegistrationTeamTest.java [hunk 4]
```diff
@@ -5,34 +5,72 @@
...
-import static org.junit.jupiter.api.Assertions.assertNotEquals;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.example.utils.Constants.*;
+import static org.junit.jupiter.api.Assertions.*;
 
 import org.example.utils.ExcelUtils;
 
 public class RegistrationTeamTest extends BaseTest {
 
     @Test
```
Comment (topsycreed) [inline src/test/java/RegistrationTeamTest.java:17]: Also please add @DisplayName annotation to add some test name that will be shown in Allure reports

File: src/test/java/ErrorValidationTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import basecomponents.BaseTest;
+import org.example.pageobjects.RegisterGamePage;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ErrorValidationTest extends BaseTest {
+
+    @Test
```
Comment (topsycreed) [inline src/test/java/ErrorValidationTest.java:9]: Probably some of your test could use 'Boundary Value Analysis and Equivalence Partitioning' test design techniques when you can pass several tests data to the same scenario to check for example that you will have error with empty phone as well as with phone with only 1 letter. To not duplicate the same scenario twice and more time use @ParameterizedTest and @ValueSource or @CsvSource or @MethodSource

Comment (topsycreed) [reply src/test/java/ErrorValidationTest.java:9]: See: https://www.baeldung.com/parameterized-tests-junit-5
